ensime_shared/ensime.py

```diff
@@ -195,15 +194,20 @@ def lazy_initialize_ensime():
             if not self.ensime:
                 called_by = inspect.stack()[4][3]
                 self.log(str(inspect.stack()))
-                self.log("setup({}, {}) called by {}()"
+                self.log("setup(quiet={}, create_classpath={}) called by {}()"
                          .format(quiet, create_classpath, called_by))
-                no_classpath = self.launcher.no_classpath_file(
-                    self.config_path)
+                no_classpath = not os.path.exists(self.launcher.classpath_file)
                 if not create_classpath and no_classpath:
                     if not quiet:
                         self.message("warn_classpath")
                     return False
-                self.ensime = self.launcher.launch(self.config_path)
+
+                try:
+                    self.ensime = self.launcher.launch()
+                except InvalidJavaPathError:
+                    # TODO: also disable plugin/shut down server here
+                    self.message('invalid_java')
+
```

Comment:
I tried doing `self.disable_plugin()` here, but it isn't really very useful—see #294.

The behavior at this point is that it will display the message prompting the user to press a key to acknowledge. That part is good. It possibly requires a key press multiple times initially, not entirely sure why but I presume it might have to do with connection retry logic causing a few messages to be buffered by Vim at one time. Then the user can quit cleanly and fix the config problem, or go on using Vim without ENSIME while the plugin log and message area continue to be spammed with the error over and over (at least non-intrusively as far as not requiring a key press) and Vim performance pays the price of futile autocommand event handler callbacks continuing to be tried.

As-is this is still a user experience improvement for #280, and I believe these caveats will be resolved by addressing #294 and/or some review of logic that retries `setup()` indefinitely regardless of a retry counter (e.g. `CursorMoved` handler).